packages/ui/react-ui-geo/src/components/Map/Map.tsx

```diff
@@ -84,28 +94,46 @@ const MapCanvas = forwardRef<MapController, MapCanvasProps>(({ markers, center,
     }
   }, [width, height]);
 
-  // Position.
-  useEffect(() => {
-    if (center) {
-      map.setView(center, zoom);
-    } else if (zoom !== undefined) {
-      map.setZoom(zoom);
-    }
-  }, [center, zoom]);
-
   // Events.
   useEffect(() => {
     const handler = debounce(() => {
-      onChange?.({ center: map.getCenter(), zoom: map.getZoom() });
+      onChange?.({
+        center: map.getCenter(),
+        zoom: map.getZoom(),
+      });
     }, 100);
+
     map.on('move', handler);
     map.on('zoom', handler);
+    map.on('focus', () => setHasFocus(true));
```

Comment:
This is a risky pattern – other components handle this using attention to determine what to do with wheel and pointer events, which I recommend doing here.

Reply:
+1